extension/src/types.ts

```diff
@@ -85,3 +85,64 @@ export enum MessageType {
     DEBUG_MESSAGE,
     CHANGE_ROOM,
 }
+
+/**
+ * Video status for one user
+ */
+enum UserVideoStatus {
+    Unknown, // Still Loading of DOM, or not on the video URL, or any othe reason (no info from Video API)
+    Buffering, // has not reached minBufferLength (video is paused or not started)
+    Playing, // is currently playing the video
+    Ready, // (equivalent of Waiting or Paused) has buffered enough of the video, is pending start signal
+}
+
+/**
+ * Video status for all players that have completed initial Sync
+ */
+enum VideoSyncStatus {
+    Paused, // video is paused and no action should be taken (apart from buffering)
+    Waiting, // waiting for all players to be in waiting status before playing
+    Playing, // video should playing normally on all user browsers
+}
+
+/**
+ * Information about one user
+ */
+export class User {
+    id: string; // maybe not needed since it will be the index key
+    connection: string; // The kind of info we need to be able to communicate with the user (TOCONFIRM: just a string is enough ?)
+    joinedAt: number;
```

Comment:
I'd rather store date information as an ISO string with timezone

Reply:
DynamoDB allow either storing dates as string or unix numbers, but it's easier to manage them as unix numbers (for comparisons, etc)

However, we can totally model this differently on the app side, and convert unix timestamps to dates. But for DDB storage it's better to use unix timestamps imo

Reply:
But we will have to handle timezone conversion to timestamp no ? Or wll this be gracefully handled by a lib ?

Reply:
Storing dates as ISO strings will also help us debugging if we need to look at the DB level data

Reply:
Yes it's true that debugging with unix timestamps can be quite a pain as well (I have experienced it as well) and debugging should be a priority.

Regarding the TZ, I'm not sure whether or not we'll have problems. If we convert and do everything in ut it should be fine right ? We're never interested in the actual time for the user but only whether two people or not are watching at the same moment, so we should always ban timezones (it will also protect us from all the DST krap).

The advantage of storing dates as ISO8601 is that you also have alphabetical sort possible, so sorting can still be done. What you cannot do if you store as a string, is for example filter db items based on "time is after xx". 

Actually, nothing prevents us from storing both integer and string formats in the DB (we'll always have to update both fields at the same time but it's fine). 

Since I'm not we'll need to implement such filtering operations then maybe let's go with your suggestion and store every data as string/iso8601 representation, and if needs be we can add fields to support filtering operations.